client/app/components/Paginator/RangeButtons.js

```diff
@@ -0,0 +1,57 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+// Styles
+import bulma from 'styles/bulma.scss';
+
+// Given a range of pages, render the numeric buttons for accessing respective pages
+const RangeButtons = ({ currPage, numPages, getPosts }) => {
+  const midPoint = Math.floor(numPages / 2);
+
+  let pageRange;
+  if (numPages <= 6) {
+    // Build up an array of the page indicies from 1...numPages for our li's
+    pageRange = Array(numPages).fill(0).map((e, i) => i + 1);
+  } else if (numPages > 6 && currPage <= 3) {
+    // Directly show 1-3 at this low range and show midpoint/end for sense of scale
+    pageRange = [1, 2, 3, '...', midPoint, '...', numPages];
+  } else if (numPages > 6 && currPage >= (numPages - 2)) {
+    // Likewise, show last three at this range and beginning/midpoint for sense of scale
+    pageRange = [1, '...', midPoint, '...', numPages - 2, numPages - 1, numPages];
+  } else {
+    // In all other cases, we want the current page to be focused and centered
+    pageRange = [1, '...', currPage - 1, currPage, currPage + 1, '...', numPages];
+  }
+
```

Comment:
I'll have to look deeper into enums, not sure if they'd be any 'better' in this case but good excuse to learn anyway!